crates/fluvio-smartengine/src/lib.rs

```diff
@@ -1,21 +1,9 @@
 cfg_if::cfg_if! {
     if #[cfg(feature = "engine")] {
-        pub(crate) mod memory;
-        pub(crate) mod transforms;
-        pub(crate) mod init;
-        pub(crate) mod error;
-        pub mod metrics;
         mod engine;
-        mod state;
         pub use engine::*;
-        pub mod instance;
     }
 }
```

Comment:
The new `engine` mod will correspond to the feature `engine`. The implementations are moved to `engine::wasmtime_engine`.